bulk-model-sync-lib/src/jvmTest/kotlin/org/modelix/model/sync/bulk/ModelImporterOrderPropertyTest.kt

```diff
@@ -0,0 +1,355 @@
+                    "concept": "uid1",
+                    "role": "uid3"
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 moved with because of the role change from "uid2" to "uid3".
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
+        // child1 changes its role from "uid2" to "uid3".
+        assertEquals("uid3", children[1].getContainmentLink()!!.getUID())
+    }
+
+    @Test
+    fun `unordered children with changing parent are appended`() {
+        // language=json
+        val dataBeforeImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              },
+              {
+                "id": "parent2",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataBeforeImport = ModelData.fromJson(dataBeforeImportJson)
+        val branch = loadModelIntoBranch(dataBeforeImport)
+        val modelImporter = ModelImporter(branch.getRootNode())
+
+        // language=json
+        val dataToImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2,"
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 is added after existing child2 because it was moved from another node
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
```

Comment:
Same here. If `IChildLink.isOrders == false` then we can't expect a specific order.

Reply:
Done